core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java

```diff
@@ -1523,5 +1529,31 @@ private void validateStringCharacterLimit() {
     }
   }
 
-
+  /**
+   * This method validates the DETAIL_QUERY_BATCH_SIZE. If some invalid input is set, we use the
+   * default value for this property
+   */
+  private void validateDetailQueryBatchSize() {
+    String batchSizeString =
+        carbonProperties.getProperty(CarbonCommonConstants.DETAIL_QUERY_BATCH_SIZE);
+    if (batchSizeString == null) {
+      carbonProperties.setProperty(DETAIL_QUERY_BATCH_SIZE,
+          Integer.toString(CarbonCommonConstants.DETAIL_QUERY_BATCH_SIZE_DEFAULT));
+      LOGGER.info(
+          "Using default value for carbon.detail.batch.size " + DETAIL_QUERY_BATCH_SIZE_DEFAULT);
+    } else {
+      int batchSize;
+      try {
+        batchSize = Integer.parseInt(batchSizeString);
+        if (batchSize < 100 || batchSize > 1000) {
```

Comment:
a) should not use hard coded values. Suggest to define a final variable in CCC and use it.
b) Also need to update the document about this mix max values. default value.

Reply:
Done.